src/main/java/uk/gov/pay/ledger/transaction/service/AccountIdSupplierManager.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.pay.ledger.transaction.service;
+
+import uk.gov.pay.ledger.exception.ValidationException;
+import uk.gov.pay.ledger.transaction.search.model.TransactionView;
+
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class AccountIdSupplierManager {
+
+    private static final String GATEWAY_ACCOUNT_ID = "account_id";
+
+    private final boolean overrideAccountRestriction;
+    private final String gatewayAccountId;
+
+    private Supplier<Optional<TransactionView>> privilegedSupplier;
+    private Function<String, Optional<TransactionView>> supplier;
+
+    public AccountIdSupplierManager(Boolean overrideAccountRestriction, String gatewayAccountId) {
```

Comment:
Maybe make this private since you have a public static factory

Reply:
👍